AtSheet/explainer.md

```diff
@@ -0,0 +1,232 @@
+- [Discussion forum](https://github.com/w3c/csswg-drafts/issues/5629)
+
+## Status of this Document
+
+This document is intended as a starting point for engaging the community and
+standards bodies in developing collaborative solutions fit for standardization.
+As the solutions to problems described in this document progress along the
+standards-track, we will retain this document as an archive and use this section
+to keep the community up-to-date with the most current standards venue and
+content location of future work and discussions.
+
+* This document status: **Active**
+* Expected venue: [CSS Working Group](https://www.w3.org/Style/CSS/)
+* Current version: this document
+
+## Introduction
+When developing web components, web authors often encounter challenges with distributing global styles into shadow roots and sharing styles across different shadow roots. Declarative shadow DOM (DSD) enables creation of shadow DOM without JavaScript. However, adding styles to DSD requires the developer to either use JavaScript to put a shared stylesheet into `adoptedStyleSheets`, or to duplicate the styles in a `<style>` element for each component instance.
+
+Additionally, bundling of stylesheets is difficult for developers who are distributing web components. They either need to ship many small stylesheets, or use workarounds like `@import url("data...")` which are suboptimal for performance and don't interact well with other patterns.
+
+We propose an enhancement to allow the declaration of new stylesheets via an `@sheet` CSS block and using existing mechanisims such as `@import`, `<link>`, and CSS module script `import` to apply those shared styles to DSDs without the use of JavaScript.
+
+We're currently investigating this and [Declarative CSS modules](/ShadowDOM/explainer.md) in parallel, and anticipate that we'll be prioritizing only one of these two in the immediate future.
+
+## Goals
+* Allow the reuse of styles in markup-based shadow DOM without requiring JavaScript.
+* Allow reuse of styles in markup-based shadow DOM without requiring external network requests.
+* Allow web authors to selectively pass in global styles from the parent document. 
+* Allow component authors to bundle their CSS into a single file.
+* Allow named `@sheet` references to fully integrate with existing CSS inclusion methods such as `@import` statements and `<link>` tags.
+
+
+## Non-goals
+Some developers have expressed interest in CSS selectors crossing through the Shadow DOM, as discussed in [issue 909](https://github.com/WICG/webcomponents/issues/909#issuecomment-1977487651). While this scenario is related to sharing styles with Shadow DOM elements, it is solving a different problem and should be addressed separately.
+
+## Proposal - `@sheet`
+Create a new `@sheet` CSS block, for separating style sheets with named identifiers.
+
+
+```css
+div {
+  color: blue;
+}
+
+@sheet foo {
```

Comment:
[Tab's summary of the idea](https://github.com/w3c/csswg-drafts/issues/5629#issuecomment-1405731979) included the following, which this example doesn't follow:
> `@sheet` rules have the same placement restrictions as `@import` - only at the top of a stylesheet.

Having said that, I'm a little fuzzy on the 'why' behind this restriction, so it might be good to get clarity on that.

Reply:
I *think* it's to obey this requirement:

```
Any [@import](https://drafts.csswg.org/css-cascade-5/#at-ruledef-import) rules must precede all other valid at-rules and style rules in a style sheet (ignoring [@charset](https://drafts.csswg.org/css-syntax-3/#at-ruledef-charset) and [@layer statement](https://drafts.csswg.org/css-cascade-5/#layer-empty) rules) and must not have any other valid at-rules or style rules between it and previous @import rules, or else the @import rule is invalid.
```

https://drafts.csswg.org/css-cascade-5/#at-import

Reply:
(still not sure on the "why" for that either though)